### PR TITLE
added highlight, fade and delete option

### LINKED ISSUE #54: Delete a Bar on Long Click after showing an Alert Dialogue.
**Describe the solution you'd like**
A bar will get deleted by clicking on "yes" in an Alert dialogue Box which will appear after long Click on the selected Bar.
I want to work on this issue.

### PR DESCRIPTION
**Pull request by Alaska18**
## Issue that this pull request solves

Fixes: #54 and #74 

## Proposed changes

kindly consider this PR for fixing issues #54 and #74. The PR before this got reset.

### CODE REVIEW

**Comment by Alaska18** [comment]
these are the images

File: library/src/main/java/in/krharsh17/barview/BarGroup.java [hunk 1]
```diff
@@ -582,12 +575,250 @@ public boolean willChangeBounds() {
...
         GradientDrawable gradientDrawable = new GradientDrawable();
         gradientDrawable.setShape(GradientDrawable.RECTANGLE);
-        gradientDrawable.setColor(color);
         gradientDrawable.setCornerRadius(CORNER_RADIUS);
+        gradientDrawable.setStroke(8, highlightColor);
+        gradientDrawable.setColor(color);
         return gradientDrawable;
     }
 
+    public String changeColor(int backColor, double factor) {
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/BarGroup.java:594]: Please provide documentation

File: library/src/main/java/in/krharsh17/barview/BarGroup.java [hunk 2]
```diff
@@ -582,12 +575,250 @@ public boolean willChangeBounds() {
...
+        String modifiedColor;
+        float[] hsv = new float[3];
+        Color.colorToHSV(backColor, hsv);
+        hsv[2] *= factor;
+        modifiedColor = String.format("#%06X", (0xFFFFFF & Color.HSVToColor(hsv)));
+        return modifiedColor;
+
+    }
+
+    public void setLayerForBar(int numberOfLayers, Drawable[] layers) {
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/BarGroup.java:611]: Documentation required

File: library/src/main/java/in/krharsh17/barview/BarGroup.java [hunk 3]
```diff
@@ -582,12 +575,250 @@ public boolean willChangeBounds() {
...
+        }
+        increaseHeight = 1;
+        increaseWidth = 0;
+        if (this.numberOfLayers > 0) {
+            increaseHeight = 1.2f + (numberOfLayers - 5) * 0.1f;
+            increaseWidth = 20 + (numberOfLayers - 5) * 4;
+        }
+    }
+
+    public void setLayerForInitial(int numberOfLayers, Drawable[] layers) {
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/BarGroup.java:633]: Documentation

**Review by krharsh17 — DISMISSED** [review]
Please squash the commits @Alaska18

**Comment by Alaska18** [comment]
I tried to squash it all, but I couldn't squash the merge commit.

**Comment by Alaska18** [comment]
I think now it's fine.

**Review by krharsh17 — DISMISSED** [review]
Looks good to me, resolve the conflicts so that we may merge it.

**Review by samagragupta — COMMENTED** [review]
Please resolve the conflicts.

**Comment by Alaska18** [comment]
conflicts are now resolved I think

**Review by krharsh17 — APPROVED** [review]

**Pull request merged**